nav2_behavior_tree/include/nav2_behavior_tree/bt_action_node.hpp

```diff
@@ -134,6 +134,10 @@ class BtActionNode : public BT::ActionNodeBase
       on_tick();
 
       on_new_goal_received();
+
+      rclcpp::spin_some(node_);
```

Comment:
Can you explain why spin_some was needed here?

Reply:
I separated `on_tick` and `on_wait_for_result` in different ticks because tests assume so. Technically we can wait `spin_some` until next tick cycle, but maybe better to put this here to allow the action server starts working soon.

Reply:
I think we should wait until after we leave tick, we don't want tick to hang if processing callbacks for other BT nodes in tick.